packages/x-date-pickers-pro/src/DateRangeCalendar/DateRangeCalendar.tsx

```diff
@@ -277,10 +277,9 @@ const DateRangeCalendar = React.forwardRef(function DateRangeCalendar<
         onRangePositionChange(nextSelection);
       }
 
-      const isFullRangeSelected = rangePosition === 'end' && isRangeValid(utils, newRange);
       setValueAndGoToNextView(
         newRange,
-        isFullRangeSelected || !isNextSectionAvailable ? 'finish' : 'partial',
+        isRangeValid(utils, newRange) || !isNextSectionAvailable ? 'finish' : 'partial',
```

Comment:
IMHO we should call `onAccept` when you already have a range filled and you just click on one date.
Because we have no way of knowing if this action will be followed by another one, you could just click on a day to shrink your range and wait for your data to update.